### PR TITLE
EclipseRestart-improvements

### PR DESCRIPTION
**Pull request by jalvestad**
Added improvements and corrections to code for writing of Eclipse compatible restart file in order to handle more general well data (Connection and segmented well data)

### CODE REVIEW

File: src/opm/output/eclipse/AggregateConnectionData.cpp [hunk 1]
```diff
@@ -71,6 +71,8 @@ namespace {
 	for (const auto & conn : conns) {
 	    std::size_t sI = conn.getCompSegSeqIndex();
 	    cs_seqIndConnMap.insert(std::make_pair(sI, &conn));
+	    std::cout << "mapCSSITC: connSI sI: " << sI <<  " I: " << conn.getI() << " J: " << conn.getJ()
```
Comment (joakim-hove) [inline src/opm/output/eclipse/AggregateConnectionData.cpp]: This looks like a debug message?

Comment (jalvestad) [reply src/opm/output/eclipse/AggregateConnectionData.cpp]: I think all the debug statements are removed in the final version of the code!?

Comment (joakim-hove) [reply src/opm/output/eclipse/AggregateConnectionData.cpp]: Yes you are right - I looked at the first commit.

**Comment by bska** [comment]
Since this enables XCON output we will have to "update_data" before merging.  I'll run a regular Jenkins build to assess the situation.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
I think this should go in&mdash;with updated reference solutions&mdash;so that we're better prepared for general well topologies in the context of restarted simulations.

Any objections?  If not, I'll probably merge this around noon CEST today (Tuesday 2nd of October).

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=63 please

**Comment by bska** [comment]
I'm merging this into master now as promised.

**Pull request merged**